index.js

```diff
@@ -71,7 +75,24 @@ function prebuildify (opts, cb) {
     if (err) return cb(err)
     loop(opts, function (err) {
       if (err) return cb(err)
-
+      if (opts.optionalPackages) {
+        var description = 'Platform specific binary for ' + packageData.name + ' on ' + opts.platform + ' OS with ' + opts.arch + ' architecture'
+        fs.writeFileSync(path.join(opts.builds, 'package.json'), JSON.stringify({
+          name: packageData.name + '-' + opts.platform + '-' + opts.arch,
```

Comment:
`opts.arch` can be multi-arch, like `x64+arm64` (https://github.com/prebuild/prebuildify/issues/52), which would result in an invalid package name. I suppose that scenario will need 2 packages

Reply:
I could probably replace the '+' with '-' for the package name and then assign `cpu: opts.arch.split('+')`, but I think the bigger question would be how we want to find the package from node-gyp-build. I don't know of any way to "scan" the available packages (for the tuple matching); it looks for a specifically constructed package name. I suppose I could put in a special case for x64 and arm64 that if package-os-x64 is not found, also try package-os-x64-arm64.

Or alternately we could just throw an error (like "combined architecture builds not support with optional-packages", since combined binaries is a little anathema to the goal of optional packages of reducing download/installation of unnecessary binary data), but whatever you think.

Reply:
> how we want to find the package from node-gyp-build

There's probably a way, but with a lookup cost (trial-and-error). I'd instead use two packages, so that npm and `node-gyp-build` would just pick one of them even though their content is the same. That would be a temporary solution to support universal binaries for M1 (in projects that also want to use optionalPackages) because it's easier to publish two packages than to change existing CI and binding.gyp files to produce separate binaries.

Anyway, that's all hypothetical, and this is not a blocker. Let's throw an error for now. I suggest "Optional packages do not support multi-arch values".